### PR TITLE
nixos/doc: Clarify wireless examples 

### LINKED ISSUE #66650: Wireless Documention is Unclear and Ambiguous
**Describe the bug**
The wireless documentation in the manual is unclear and ambiguous for a novice NixOS user wishing to configure a variety of possible SSID connections.

**To Reproduce**
Be a novice NixOS user wishing to configure free wireless or SSIDs with spaces and punctuatin.

**Expected behavior**
Clear examples be provided.

### PR DESCRIPTION
**Pull request by craigem**
This commits makes it clearer to a novice reader how to configure several
diferent types of SSID connections that were otherwise obscurely documented

Resolves #66650

###### Motivation for this change

Clarify wireless documentation

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: nixos/doc/manual/configuration/wireless.xml [hunk 1]
```diff
@@ -19,10 +19,13 @@
...
 <xref linkend="opt-networking.wireless.networks"/> = {
-  echelon = {
+  echelon = {                # SSID with no spaces or special characters
     psk = "abcdefgh";
   };
-  "free.wifi" = {};
+  "echelon's AP" = {         # SSID with spaces and/or special characters
+    psk = "ijklmnop";
+  };
+  freeWiFi = {};             # An example of how to configure open SSIDs
```
Comment (worldofpeace) [inline nixos/doc/manual/configuration/wireless.xml]:
```suggestion
  freeWiFi = {};             # Public wireless network
```

**Review by worldofpeace — CHANGES_REQUESTED** [review]
How about an example for a hidden network?

Also, should the example in the actual NixOS option be updated?

File: nixos/doc/manual/configuration/wireless.xml (hunk 1 above)
Comment (craigem) [reply nixos/doc/manual/configuration/wireless.xml]: Great suggestions. Updated accordingly.

File: nixos/modules/services/networking/wpa_supplicant.nix [hunk 2]
```diff
@@ -146,10 +153,13 @@ in {
...
         example = literalExample ''
-          { echelon = {
+          { echelon = {                   # SSID with no spaces or special characters
               psk = "abcdefgh";
             };
-            "free.wifi" = {};
+            "echelon's AP" = {            # SSID with spaces and/or special characters
+               psk = "ijklmnop";
+            };
+            "free.wifi" = {};             # Public wireless network
```
Comment (worldofpeace) [inline nixos/modules/services/networking/wpa_supplicant.nix]:
```suggestion
            freeWifi = {};             # Public wireless network
```

So it's the same as the manual.

**Review by worldofpeace — CHANGES_REQUESTED** [review]
Can we have the example in the manual have the same ap names as the options?

So the hidden example in the option has `echelon =` while in the manual it's `echelonHidden`.
Similar with the `free.wifi` and `freeWiFi`.

**Comment by worldofpeace** [comment]
Thanks @craigem :sparkles:

**Pull request merged**